event_callback.go

```diff
@@ -0,0 +1,170 @@
+package firego
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/zabawaba99/firego/sync"
+)
+
+// ChildEventFunc is the type of function that is called for every
+// new child added under a firebase reference. The snapshot argument
+// contains the data that was added. The previousChildKey argument
+// contains the key of the previous child that this function was called for.
+type ChildEventFunc func(snapshot DataSnapshot, previousChildKey string)
+
+// ChildAdded listens on the firebase instance and executes the callback
+// for every child that is added.
+func (fb *Firebase) ChildAdded(fn ChildEventFunc) error {
+	handleSSE := func(notifications chan Event) {
+		first, ok := <-notifications
+		if !ok {
+			return
+		}
+
+		var pk string
+		db := sync.NewDB()
+		children, ok := first.Data.(map[string]interface{})
+		if ok {
+			// we've got children so send an event per child
+			orderedChildren := make([]string, len(children))
+			var i int
+			for k := range children {
+				orderedChildren[i] = k
+				i++
+			}
+
+			sort.Strings(orderedChildren)
+
+			for _, k := range orderedChildren {
+				v := children[k]
+				node := sync.NewNode(k, v)
+				db.Add(k, node)
+				fn(newSnapshot(node), pk)
+				pk = k
+			}
+		}
+
+		for event := range notifications {
+			if event.Type != "put" {
```

Comment:
I'm not too sure I see the value in adding a constant just to compare here. I wrote some benchmarks to see if it would make a big difference but it didn't seem to (at least on go1.6). 

On a side note - I could see some benefit in making those exported typed that other people could use when using `Watch` directly.